output/cloud/expv2/metrics_client.go

```diff
@@ -0,0 +1,125 @@
+package expv2
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/golang/snappy"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/proto"
+
+	"go.k6.io/k6/lib/consts"
+	"go.k6.io/k6/output/cloud/expv2/pbcloud"
+)
+
+type httpDoer interface {
+	Do(*http.Request) (*http.Response, error)
+}
+
+// MetricsClient is a Protobuf over HTTP client for sending
+// the collected metrics from the Cloud output
+// to the remote service.
+type MetricsClient struct {
+	httpClient httpDoer
+	logger     logrus.FieldLogger
+	token      string
+	userAgent  string
+
+	pushBufferPool sync.Pool
+	baseURL        string
+}
+
+// NewMetricsClient creates and initializes a new MetricsClient.
+func NewMetricsClient(logger logrus.FieldLogger, host string, token string) (*MetricsClient, error) {
+	if host == "" {
+		return nil, errors.New("host is required")
+	}
+	if token == "" {
+		return nil, errors.New("token is required")
+	}
+	return &MetricsClient{
+		httpClient: &http.Client{Timeout: 5 * time.Second},
+		logger:     logger,
+		baseURL:    host + "/v2/metrics/",
+		token:      token,
+		userAgent:  "k6cloud/v" + consts.Version,
```

Comment:
Isn't the `consts.Version` already has a `v` prefix? :thinking:

Reply:
Nope: https://github.com/grafana/k6/blob/4e8347ee4b12c65069e55e7c7fe82ec9aa57d777/lib/consts/consts.go#L11

Reply:
Right :facepalm: :+1: